build/lib/src/generate/run_builder.dart

```diff
@@ -24,7 +24,8 @@ Future<Null> runBuilder(Builder builder, Iterable<AssetId> inputs,
     AssetReader reader, AssetWriter writer, Resolvers resolvers,
     {Logger logger, String rootPackage}) async {
   logger ??= new Logger('runBuilder');
-  rootPackage ??= new Set.from(inputs.map((input) => input.package)).single;
+  rootPackage ??=
+      new Set<String>.from(inputs.map((input) => input.package)).single;
```

Comment:
The signature of `Set.from` is `Set.from(Iterable elements)` - note the implicit `<dynamic>` and that the generic arg on `Set` of `<E>` doesn't appear in the signature. There is nothing for the ctor to use to infer the type for the constructed instance.

The reason is that the `from` constructors are intended to help you get a collection with the reified type you want if you know all the elements are of the correct type, even if the reified type of the input Iterable doesn't match. If the signature were `Set.from(Iterable<E> elements)` then some other usages would get more clunky...

`new Set.from(stuff.where((e) => e is String).map((e) => e as String);`